st2client/st2client/commands/action.py

```diff
@@ -277,8 +277,65 @@ class ActionDisableCommand(resource.ContentPackResourceDisableCommand):
     ]
 
 
-class ActionDeleteCommand(resource.ContentPackResourceDeleteCommand):
-    pass
+class ActionDeleteCommand(resource.ResourceCommand):
+    pk_argument_name = "ref_or_id"
+
+    def __init__(self, resource, *args, **kwargs):
+        super(ActionDeleteCommand, self).__init__(
+            resource,
+            "delete",
+            "Delete an existing %s." % resource.get_display_name().lower(),
+            *args,
+            **kwargs,
+        )
+
+        argument = self.pk_argument_name
+        metavar = self._get_metavar_for_argument(argument=self.pk_argument_name)
+        help = self._get_help_for_argument(
+            resource=resource, argument=self.pk_argument_name
+        )
+
+        self.parser.add_argument(argument, metavar=metavar, help=help)
```

Comment:
Why are lines 281, 292-298 here needed? If you change the base class that ActionDeleteCommand inherits from to resource.ResourceDeleteCommand, the __init__ can be simplified.

Reply:
Try the following instead

```
class ActionDeleteCommand(resource.ResourceDeleteCommand):
    def __init__(self, resource, *args, **kwargs):
        super(ActionDeleteCommand, self).__init__(resource, *args, **kwargs)

        self.parser.add_argument(
            "-f",
            action="store_true",
            help="Auto yes flag to delete action files from disk.",
        )

        self.parser.add_argument(
            "--force",
            action="store_true",
            help="Auto yes flag to delete action files from disk.",
        )
```